src/main/java/mate/academy/hibernate/relations/dao/impl/ActorDaoImpl.java

```diff
@@ -12,11 +15,29 @@
 
     @Override
     public Actor add(Actor actor) {
-        return null;
+        Session session = factory.openSession();
+        if (session == null) {
+            throw new DataProcessingException("Session not initialized");
+        }
+        Transaction transaction = session.beginTransaction();
+        try {
+            session.persist(actor);
+            transaction.commit();
+        } catch (Exception e) {
+            transaction.rollback();
+            throw new DataProcessingException("Error in adding actor", e);
+        }
+        return actor;
```

Comment:
You should not return the same entity instance that you've passed to `persist` method. Instead, return a new instance retrieved from the database to ensure that you're returning the entity with generated ID and any other changes that might have occurred during the persisting process.